packages/dht/src/dht/contact/RandomContactList.ts

```diff
@@ -50,13 +50,6 @@ export class RandomContactList<C extends { getNodeId: () => DhtAddress }> extend
     }
 
     public getContacts(limit = this.defaultContactQueryLimit): C[] {
-        const ret: C[] = []
-        this.contactIds.forEach((contactId) => {
-            const contact = this.contactsById.get(contactId)
-            if (contact) {
-                ret.push(contact.contact)
-            }
-        })
-        return ret.slice(0, limit)
+        return this.contactIds.map((contactId) => this.contactsById.get(contactId)!).slice(0, limit)
```

Comment:
I would slice first for performance
```suggestion
        return this.contactIds.slice(0, limit).map((contactId) => this.contactsById.get(contactId)!)
```